pkg/mutation/system.go

```diff
@@ -100,12 +106,45 @@ func (s *System) Mutate(obj *unstructured.Unstructured, ns *corev1.Namespace) (b
 			if cmp.Equal(original, obj) {
 				return false, fmt.Errorf("oscillating mutation for %s %s", obj.GroupVersionKind().Group, obj.GroupVersionKind().Kind)
 			}
+			logAppliedMutations(original, allAppliedMutations)
 			return true, nil
 		}
+		allAppliedMutations = append(allAppliedMutations, appliedMutations)
 	}
 	return false, fmt.Errorf("mutation not converging for %s %s", obj.GroupVersionKind().Group, obj.GroupVersionKind().Kind)
 }
 
+func logAppliedMutations(obj *unstructured.Unstructured, allAppliedMutations [][]types.Mutator) {
+	appliedMutationsText := ""
+	for i, appliedMustations := range allAppliedMutations {
+		if len(appliedMustations) == 0 {
+			continue
+		}
+		if i != 0 {
+			appliedMutationsText += "; "
+		}
+		appliedMutationsText += fmt.Sprintf("Iteration %d:", i)
+		for j, mutator := range appliedMustations {
+			if j != 0 {
+				appliedMutationsText += ", "
+			}
+			mutatorName := mutator.ID().Name
+			if len(mutator.ID().Namespace) > 0 {
+				mutatorName = fmt.Sprintf("%s/%s", mutator.ID().Namespace, mutatorName)
```

Comment:
Do we also want to print out the kind so we can disambiguate if an Assign and AssignMetadata have the same name?

Reply:
Added kind along with resource version